File: Data Structures & Algorithms/binary-search/submission-1.cpp
class Solution {
public:
    int search(vector<int>& arr, int target) {
        int st=0,end=arr.size()-1;
        while(st<=end){
            int mid=(st+end)/2;
            if(arr[mid]==target){
                return mid;
            }
            else if(arr[mid]<target){
                st=mid+1;
            }
            else{
                end=mid-1;
            }
        }
        return -1;
    }
};

File: Data Structures & Algorithms/car-fleet/submission-1.py
class Solution:
    def carFleet(self, target: int, position: List[int], speed: List[int]) -> int:
        number_of_fleets = 0
        last_fleet_time = 0
        for pos, spd in sorted(zip(position, speed), reverse=True):
            time_taken = (target-pos) / spd
            if time_taken > last_fleet_time:
                last_fleet_time = time_taken
                number_of_fleets += 1
        return number_of_fleets